// @x402/evm/dist/esm/index.mjs
import {
  BatchSettlementEvmScheme
} from "./chunk-PHFSKO67.mjs";
import "./chunk-W6ON4LG2.mjs";
import {
  BATCH_SETTLEMENT_ADDRESS,
  BATCH_SETTLEMENT_DOMAIN,
  BATCH_SETTLEMENT_SCHEME,
  ERC3009_DEPOSIT_COLLECTOR_ADDRESS,
  claimBatchTypes,
  refundTypes,
  voucherTypes
} from "./chunk-HYABYUBD.mjs";
import {
  ExactEvmScheme
} from "./chunk-64DPS35M.mjs";
import {
  DEFAULT_STABLECOINS,
  getDefaultAsset
} from "./chunk-BTYNCDNS.mjs";
import {
  isEIP3009Payload,
  isPermit2Payload,
  isUptoPermit2Payload
} from "./chunk-JK7SLLF7.mjs";
import {
  isBatchSettlementClaimPayload,
  isBatchSettlementDepositPayload,
  isBatchSettlementEnrichedRefundPayload,
  isBatchSettlementRefundPayload,
  isBatchSettlementSettlePayload,
  isBatchSettlementVoucherPayload
} from "./chunk-U4HCGTLU.mjs";
import "./chunk-TRL63EE2.mjs";
import {
  UptoEvmScheme
} from "./chunk-6WQOGWBE.mjs";
import {
  createPermit2ApprovalTx,
  getPermit2AllowanceReadParams
} from "./chunk-T55QNCK7.mjs";
import "./chunk-YMQCTKDU.mjs";
import "./chunk-5XCWSAME.mjs";
import "./chunk-GD4MKCN7.mjs";
import "./chunk-ZZV4WOUI.mjs";
import {
  PERMIT2_ADDRESS,
  authorizationTypes,
  eip3009ABI,
  erc20AllowanceAbi,
  permit2WitnessTypes,
  uptoPermit2WitnessTypes,
  x402ExactPermit2ProxyABI,
  x402ExactPermit2ProxyAddress,
  x402UptoPermit2ProxyABI,
  x402UptoPermit2ProxyAddress
} from "./chunk-MACPBXCT.mjs";
import "./chunk-VS3RYAYE.mjs";
import "./chunk-TW7Z65AO.mjs";

// src/signer.ts
function toClientEvmSigner(signer, publicClient) {
  const readContract = signer.readContract ?? publicClient?.readContract.bind(publicClient);
  const result = {
    address: signer.address,
    signTypedData: (msg) => signer.signTypedData(msg)
  };
  if (readContract) {
    result.readContract = readContract;
  }
  const signTransaction = signer.signTransaction;
  if (signTransaction) {
    result.signTransaction = (args) => signTransaction(args);
  }
  const getTransactionCount = signer.getTransactionCount ?? publicClient?.getTransactionCount?.bind(publicClient);
  if (getTransactionCount) {
    result.getTransactionCount = (args) => getTransactionCount(args);
  }
  const estimateFeesPerGas = signer.estimateFeesPerGas ?? publicClient?.estimateFeesPerGas?.bind(publicClient);
  if (estimateFeesPerGas) {
    result.estimateFeesPerGas = () => estimateFeesPerGas();
  }
  return result;
}
function toFacilitatorEvmSigner(client) {
  return {
    ...client,
    getAddresses: () => [client.address]
  };
}
export {
  BATCH_SETTLEMENT_ADDRESS,
  BATCH_SETTLEMENT_DOMAIN,
  BATCH_SETTLEMENT_SCHEME,
  BatchSettlementEvmScheme,
  DEFAULT_STABLECOINS,
  ERC3009_DEPOSIT_COLLECTOR_ADDRESS,
  ExactEvmScheme,
  PERMIT2_ADDRESS,
  UptoEvmScheme,
  authorizationTypes,
  claimBatchTypes,
  createPermit2ApprovalTx,
  eip3009ABI,
  erc20AllowanceAbi,
  getDefaultAsset,
  getPermit2AllowanceReadParams,
  isBatchSettlementClaimPayload,
  isBatchSettlementDepositPayload,
  isBatchSettlementEnrichedRefundPayload,
  isBatchSettlementRefundPayload,
  isBatchSettlementSettlePayload,
  isBatchSettlementVoucherPayload,
  isEIP3009Payload,
  isPermit2Payload,
  isUptoPermit2Payload,
  permit2WitnessTypes,
  refundTypes,
  toClientEvmSigner,
  toFacilitatorEvmSigner,
  uptoPermit2WitnessTypes,
  voucherTypes,
  x402ExactPermit2ProxyABI,
  x402ExactPermit2ProxyAddress,
  x402UptoPermit2ProxyABI,
  x402UptoPermit2ProxyAddress
};
